phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,255 @@
+     * objects toString().
+     */
+    public String serializeToString() {
+        if (this.rootNode == null || this.rootNode.isNull()) {
+            return null;
+        } else if (this.rootNode.isValueNode()) {
+
+            if (this.rootNode.isNumber()) {
+                return this.rootNode.getNumberValue().toString();
+            } else if (this.rootNode.isBoolean()) {
+                return String.valueOf(this.rootNode.getBooleanValue());
+            } else if (this.rootNode.isTextual()) {
+                return this.rootNode.getTextValue();
+            } else {
+                return getJsonAsString();
+            }
+        } else if (this.rootNode.isArray()) {
+            return getJsonAsString();
+        } else if (this.rootNode.isContainerNode()) {
+            return getJsonAsString();
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public String toString() {
+        return getJsonAsString();
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((this.rootNode == null) ? 0 : this.rootNode.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        PhoenixJson other = (PhoenixJson) obj;
+        if ((this.jsonAsString != null) && (other.jsonAsString != null)) {
+            if (this.jsonAsString.equals(other.jsonAsString)) return true;
+        }
+        if (this.rootNode == null) {
+            if (other.rootNode != null) return false;
+        } else if (!this.rootNode.equals(other.rootNode)) return false;
+        return true;
+    }
+
+    /**
+     * @return length of the string represented by the current {@link PhoenixJson}.
+     */
+    public int estimateByteSize() {
+        String jsonStr = getJsonAsString();
+        return jsonStr == null ? 1 : jsonStr.length();
+    }
+
+    public byte[] toBytes() {
+        return Bytes.toBytes(getJsonAsString());
+    }
+
+    @Override
+    public int compareTo(PhoenixJson o) {
+        if (o == null) {
+            return 1;
+        } else if (this.equals(o)) {
+            return 0;
+        } else {
+            return this.toString().compareTo(o.toString());
+        }
+    }
+
+    private PhoenixJson getPhoenixJsonInternal(String[] paths) {
+        JsonNode node = this.rootNode;
+        for (String path : paths) {
+            JsonNode nodeTemp = null;
+            if (node.isArray()) {
+                int index = Integer.parseInt(path);
+                nodeTemp = node.path(index);
+            } else {
+                nodeTemp = node.path(path);
+            }
+            if (nodeTemp == null || nodeTemp.isMissingNode()) {
+                return null;
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node, null);
```

Comment:
Might be better to just call node.toString() here instead of passing in null. Otherwise, the getJsonAsString() and toString() methods will allocate a big string on each call.